kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -222,7 +223,9 @@ public T getMandatory() {
     if (item != null) {
       return item;
     }
-    throw new KubernetesClientException("Resource not found!", 404, new StatusBuilder().withCode(404).withMessage("Not found.").build());
```

Comment:
@jimmidyson: We can modify the `KubernetesClientException` in order to accept an additional object, that contains (name, namespace, kind, and type of opertation). The 3 first are available to all operations as the are part of `BaseOperation` the type of operation is method specific and needs to be passed when we call the `launderException`.

This can work and in fact I have implemented something like this locally and plays nicely. However, this requires a minor version bump. So how about we merge it as is, for 2.2.x and I'll implement it as described for 2.3.